Source/Dafny/Compilers/Compiler-js.cs

```diff
@@ -2401,6 +2404,17 @@ protected class ClassWriter : IClassWriter {
         }
         nodeProcess.StandardInput.Flush();
         nodeProcess.StandardInput.Close();
+        // Fixes a problem of Node on Windows, where Node does not prints to the parent console its standard outputs.
+        Task.Run(() => {
+          int current;
+          while ((current = nodeProcess.StandardError.Read()) != -1) {
+            Console.Error.Write((char)current);
+          }
+        });
+        int current;
+        while ((current = nodeProcess.StandardOutput.Read()) != -1) {
+          Console.Out.Write((char)current);
+        }
```

Comment:
Looks good now, though reading character-by-character is a bit scary.  I'm not sure it's that important since it's just a debugging facility, but if it is I would spawn two instances of a class that does something roughly like 

```
while (var count = await ReadAsync(buf, 0, bufsize)) {
  outputStream.Write(buf)
}
```

IIRC `ReadAsync` is only blocks until some output is received.

Reply:
I second that it might be beneficial to not read char by char. However, since the Dafny Compiler consists purely of synchronous methods, I suggest using the non-async version instead.
Either use `Read` with a buffer or `ReadLine`. Although, the latter might be not the best choice since we cannot clearly say that the output is always separated by lines.

Reply:
About character by character vs. buffering.
Buffering is a best practice when the goal is to recover the entire output into some kind of file, or for real-time application processing. Not when it's user-facing.
Here, we are talking about `dafny.exe -compile:3 -compileTarget:js`, which is a tool mostly used while developing a JavaScript application in Dafny. Its output is mean to be visible by the user, for example in the IDE, else they would just use `node` on a generated file.
So now, if the output has I/O operations, buffering the output would result in not displaying in real-time the output, which would result in a disastrous user experience.

Reply:
I refactored the code so that it's now inside a function